### ISSUE TITLE
Compiler error on dynamic import

### ISSUE THREAD

**Issue by jakzo** [thheller/shadow-cljs#486]
[JS supports using `import` as a function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports). When shadow-cljs encounters one of these, the Closure compiler fails.

## Example

`:require`ing a JS file with the following code:

```js
export default function() {
  import("hello").then(function(mod) {
    alert(mod.default);
  });
}
```

will fail when compiling with this error:

```sh
shadow-cljs - config: /.../shadow-dynamic-import-issue/shadow-cljs.edn  cli version: 2.7.17  node: v11.6.0
shadow-cljs - starting ...
[:dynamic] Compiling ...
errors in file: /.../shadow-dynamic-import-issue/node_modules/dynamic-import/index.js
{:js-str-offsets [], :js-esm false, :js-imports [], :js-invalid-requires [], :goog-provides [], :js-language "es3", :goog-module nil, :js-warnings [], :resource-name "node_modules/dynamic-import/index.js", :js-requires [], :js-errors [{:line 2, :column 2, :message "'}' expected"}], :goog-requires [], :tag :shadow.build.npm/errors, :uses-global-buffer false}
ExceptionInfo: errors in file: /.../shadow-dynamic-import-issue/node_modules/dynamic-import/index.js
        clojure.core/ex-info (core.clj:4739)
        clojure.core/ex-info (core.clj:4739)
        shadow.build.npm/get-file-info* (npm.clj:530)
        ...
```

Repro: https://github.com/jakzo/shadow-dynamic-import-issue

**Comment by thheller** [comment]
Dynamic `import` is not currently supported by the Closure Compiler (or `shadow-cljs`).

Some initial support was added [recently](https://github.com/google/closure-compiler/pull/3204) but I'm not yet sure how final support will look. I don't think this is a good idea to begin with so we'll see how it turns out.

**Comment by FalseProtagonist** [comment]
Sorry, just to check that I'm interpreting these issues correctly - it's now supported by Google Closure but not planned to be supported in shadow-cljs (?)

**Comment by thheller** [comment]
Depends. I have not tested enough yet. Currently blocked by ClojureScript being incompatible with the latest Closure Compiler.

**Comment by gacosta89** [comment]
Hi, seems dynamic imports are supported by the Closure Compiler https://github.com/google/closure-compiler/wiki/JS-Modules#dynamic-import-expressions . But not working for me actually. Whats the status? can I help?

**Comment by thheller** [comment]
Please open a new ticket with a reproducible repo for something that may not work. This is not exactly a "small topic". So, without knowing what are trying to do I can't comment on the status.

**Comment by gacosta89** [comment]
Issue created https://github.com/thheller/shadow-cljs/issues/1084 thanks for your response.
